File: Java/TP3_MET_JAVA/src/Iteratif/ToLowerCaseIteratif.java
package Iteratif;

public class ToLowerCaseIteratif implements VisiteurMotIteratif<String> {

    @Override
    public String casVide() {
        return "";
    }

    @Override
    public String casCaractereMot(Character c, String m) {
        return m + Character.toLowerCase(c);
    }
}

File: TypeScript/TP3_MET_TS/src/Iteratif/ToLowerCaseIteratif.ts
import {VisiteurMotIteratif} from "./VisiteurMotIteratif";

export class ToLowerCaseIteratif implements VisiteurMotIteratif<string> {

    public casVide(): string {
        return "";
    }

    public casCaractereMot(c: string, m: string): string {
        return m + c.toLowerCase();
    }
}
